reporters/kamon-opentelemetry/src/main/scala/kamon/otel/SpanConverter.scala

```diff
@@ -15,212 +15,122 @@
-      .setTraceId(traceIdToByteString(span.trace.id))
-      .setSpanId(spanIdToByteString(span.id))
-      .setName(span.operationName)
-      .setKind(toProtoKind(span.kind))
-      .setStartTimeUnixNano(toEpocNano(span.from))
-      .setEndTimeUnixNano(toEpocNano(span.to))
-      .setStatus(getStatus(span))
-      .addAllAttributes(attributes.asJava)
-      .addAllLinks(links.asJava)
-      .addAllEvents(events.asJava)
-
-    //TODO add set traceState once we have something to set. Need w3c context
-    //It is a trace_state in w3c-trace-context format: https://www.w3.org/TR/trace-context/#tracestate-header
-    //See also https://github.com/w3c/distributed-tracing for more details about this field.
-    //builder.setTraceState()
-
-    //add optional parent
-    val parentId = span.parentId
-    if(!parentId.isEmpty) builder.setParentSpanId(spanIdToByteString(parentId))
+  private[otel] def getTraceFlags(sample: Boolean): TraceFlags =
+    if (sample) TraceFlags.getSampled else TraceFlags.getDefault
 
-    builder.build()
+  private[otel] def mkSpanContext(sample: Boolean, traceId: Identifier, spanId: Identifier): SpanContext = {
+    val paddedTraceId = if (traceId.string.length == 16) s"0000000000000000${traceId.string}" else traceId.string
+    SpanContext.create(paddedTraceId, spanId.string, getTraceFlags(sample), TraceState.getDefault)
   }
 
-  /**
-   * Converts the instant to EPOC nanos
-   * @param instant
-   * @return
-   */
-  private[otel] def toEpocNano(instant:Instant):Long = TimeUnit.NANOSECONDS.convert(instant.getEpochSecond, TimeUnit.SECONDS) + instant.getNano
-
-  /**
-   * Creates a proto ''Status'' of the Kamon span status
-   * @param span
-   * @return
-   */
-  private[otel] def getStatus(span:Span.Finished):Status = {
-    //according to the spec the deprecated code needs to be set for backwards compatibility reasons
-    //https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto
-    val (status, deprecatedStatus) = if(span.hasError) (Status.StatusCode.STATUS_CODE_ERROR_VALUE, Status.DeprecatedStatusCode.DEPRECATED_STATUS_CODE_UNKNOWN_ERROR_VALUE) else (Status.StatusCode.STATUS_CODE_OK_VALUE, Status.DeprecatedStatusCode.DEPRECATED_STATUS_CODE_OK_VALUE)
-    val builder = Status.newBuilder()
-      .setCodeValue(status)
-      .setDeprecatedCodeValue(deprecatedStatus)
-
-    //if there is an error message in the span we add it as status message
-    //spec states: "A developer-facing human readable error message."
-    //https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto
-    span.tags.get(Lookups.option(TagKeys.ErrorMessage)).foreach(builder.setMessage)
+  private[otel] def getStatus(span: Span.Finished): StatusData =
+    if (span.hasError) {
+      val msg = span.tags.get(Lookups.option(TagKeys.ErrorMessage)) getOrElse ""
+      StatusData.create(StatusCode.ERROR, msg)
+    } else StatusData.ok()
```

Comment:
This probably should change to `unset`, according to [the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status):
![image](https://user-images.githubusercontent.com/1302854/160603802-14ccbd09-4d57-4fc1-9dfa-cd41db390503.png)

I'm not super sure about it, or whether this is an important thing or not. I did notice that in some example apps running with the OTel instrumentation the Spans stay with UNSET status, even though they completed successfully.